src/Components/WebAssembly/WebAssembly/src/Prerendering/WebAssemblyComponentParameterDeserializer.cs

```diff
@@ -75,15 +81,21 @@ public ParameterView DeserializeParameters(IList<ComponentParameter> parametersD
 
     [DynamicDependency(JsonSerialized, typeof(ComponentParameter))]
     [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", Justification = "The correct members will be preserved by the above DynamicDependency.")]
-    // This should use JSON source generation
     public static ComponentParameter[] GetParameterDefinitions(string parametersDefinitions)
     {
-        return JsonSerializer.Deserialize<ComponentParameter[]>(parametersDefinitions, WebAssemblyComponentSerializationSettings.JsonSerializationOptions)!;
+        // Keep in sync with WebAssemblyComponentParameterDeserializerSerializerContext
+        return JsonSerializer.Deserialize<ComponentParameter[]>(parametersDefinitions, _jsonSerializerOptions)!;
     }
 
     [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "We expect application code is configured to preserve component parameter types.")]
     public static IList<object> GetParameterValues(string parameterValues)
     {
-        return JsonSerializer.Deserialize<IList<object>>(parameterValues, WebAssemblyComponentSerializationSettings.JsonSerializationOptions)!;
+        // Keep in sync with WebAssemblyComponentParameterDeserializerSerializerContext
+        return JsonSerializer.Deserialize<IList<object>>(parameterValues, _jsonSerializerOptions)!;
     }
 }
+
+[JsonSerializable(typeof(ComponentParameter[]))]
+[JsonSerializable(typeof(JsonElement))]
+[JsonSerializable(typeof(IList<object>))]
```

Comment:
You'd need to additionally register the types you expect `object` to be. Given that @MackinnonBuck is planning on adding the reflection-based resolver as a fallback it would work even if the list isn't exhaustive, but making sure the 80% is handled by the source generator should contribute to improved startup times.

Reply:
Since this `JsonSerializerContext` is only used during deserialization, we happen to know that the `object` in `IList<object>` will either be a `JsonElement` or `null`. Why we didn't initially choose to deserialize a `JsonElement[]` directly, I'm not sure.

Reply:
Why not call `Deserialize<IList<JsonElement>>` (or `Deserialize<IList<JsonElement?>>` if null is valid) and register `[JsonSerializable(typeof(IList<JsonElement?>))]` here?